thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/cache/CollectionMaxDataTimeCacheLoader.java

```diff
@@ -1,82 +1,50 @@
 package com.linkedin.thirdeye.client.cache;
 
-import java.util.ArrayList;
-import java.util.List;
-import java.util.concurrent.TimeUnit;
-
-import org.apache.helix.AccessOption;
-import org.apache.helix.PropertyPathConfig;
-import org.apache.helix.PropertyType;
-import org.apache.helix.ZNRecord;
-import org.apache.helix.manager.zk.ZNRecordSerializer;
-import org.apache.helix.manager.zk.ZkBaseDataAccessor;
-import org.apache.helix.manager.zk.ZkClient;
-import org.apache.helix.store.zk.ZkHelixPropertyStore;
-import org.apache.zookeeper.data.Stat;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.linkedin.pinot.client.ResultSetGroup;
+import com.linkedin.thirdeye.api.CollectionSchema;
+import com.linkedin.thirdeye.api.TimeSpec;
+import com.linkedin.thirdeye.client.pinot.PinotQuery;
 import com.linkedin.thirdeye.client.pinot.PinotThirdEyeClientConfig;
 
 public class CollectionMaxDataTimeCacheLoader extends CacheLoader<String, Long> {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(CollectionMaxDataTimeCacheLoader.class);
 
-  private String zkUrl;
-  private String clusterName;
-  private ZkHelixPropertyStore<ZNRecord> propertyStore;
-
-  public CollectionMaxDataTimeCacheLoader(PinotThirdEyeClientConfig pinotThirdEyeClientConfig) {
-
-   zkUrl = pinotThirdEyeClientConfig.getZookeeperUrl();
-   clusterName = pinotThirdEyeClientConfig.getClusterName();
-
-   ZkClient zkClient = new ZkClient(zkUrl);
-   zkClient.setZkSerializer(new ZNRecordSerializer());
-   zkClient.waitUntilConnected();
-
-   ZkBaseDataAccessor<ZNRecord> baseAccessor = new ZkBaseDataAccessor<>(zkClient);
-   String propertyStorePath = PropertyPathConfig.getPath(PropertyType.PROPERTYSTORE, clusterName);
+  private final LoadingCache<String, CollectionSchema> collectionSchemaCache;
+  private final LoadingCache<PinotQuery, ResultSetGroup> resultSetGroupCache;
+  private final static String COLLECTION_MAX_TIME_QUERY_TEMPLATE = "SELECT max(%s) FROM %s";
```

Comment:
better would be select max(%s) from %s where time>prev_max we have seen so far. This will apply time pruning and we would avoid going over many segments